lib/arborist/reify.js

```diff
@@ -887,6 +887,13 @@ module.exports = cls => class Reifier extends cls {
       filter: diff => diff.action === 'ADD' || diff.action === 'CHANGE',
     })
 
+    // pick up link nodes from the unchanged list as we want to run their
+    // scripts in every install despite of having a diff status change
+    for (const node of this.diff.unchanged) {
+      if (node.isLink)
+        nodes.push(node)
```

Comment:
It seems like this should be something like `node.isWorkspace` or at least, check that the realpath is within the project root.  Since we don't resolve deps on external link targets, it feels wrong to run their build scripts.

Reply:
I didn't wanted to restrict the change to just workspaces since I wanted to make sure we cover the usecase in which users are linking a local module in order to patch something (or just for development purposes) in which they want `prepare` scripts to run

Reply:
maybe:

```
      if (node.isLink && node.parent.isTop)
```

instead?

Reply:
Hm, sounds like it's something like:

```js
if (node.isLink) {
  const root = node.root.target || node.root
  if (node.target && node.target.isDescendantOf(root))
    nodes.push(node)
}
```

Reply:
Oh, no, wait, that'll just be `children` not `fsParent`.

Here's how we check for path locality in `buildIdealTree`

```js
      const external = /^\.\.(\/|$)/.test(relpath(this.path, link.realpath))
```